renderer/webtorrent.js

```diff
@@ -308,6 +315,48 @@ function getAudioMetadata (infoHash, index) {
   })
 }
 
+function selectFiles (torrentOrInfoHash, selections) {
+  // Get the torrent object
+  var torrent
+  if (typeof torrentOrInfoHash === 'string') {
+    torrent = client.get(torrentOrInfoHash)
+  } else {
+    torrent = torrentOrInfoHash
+  }
+
+  // Selections not specified?
+  // Load all files. We still need to replace the default whole-torrent
+  // selection with individual selections for each file, so we can
+  // select/deselect files later on
+  if (!selections) {
+    selections = torrent.files.map((x) => true)
+  }
+
+  // Selections specified incorrectly?
+  if (selections.length !== torrent.files.length) {
+    throw new Error('got ' + selections.length + ' file selections, ' +
+      'but the torrent contains ' + torrent.files.length + ' files')
+  }
+
+  // Remove default selection (whole torrent)
+  torrent.deselect(0, torrent.pieces.length - 1, false)
+
+  // Add selections (individual files)
+  for (var i = 0; i < selections.length; i++) {
+    var file = torrent.files[i]
+    if (selections[i]) {
+      file.select()
+    } else {
+      console.log('deselecting file ' + i + ' of torrent ' + torrent.name)
+      file.deselect()
+
+      // If we deselected a file, try to nuke it to save disk space
+      var filePath = path.join(torrent.path, file.path)
+      fs.unlink(filePath) // Ignore errors for now
```

Comment:
> ... is that possible when you're writing to both the start and end of a file?

I think that will cause the file to expand. You're probably right.

It's also worth noting that Transmission appends a `.part` extension to the end of incomplete files, which might allow them to use a format that lets them store partial files more efficiently than normal. Or that might just be a hack so that users don't try to open an incomplete .mp4 file and think Transmission is broken.